plugins/inventory_sync/module/serialize.lua

```diff
@@ -112,9 +127,46 @@ function serialize.deserialize_personal_logistic_slots(player, serialized)
 	end
 
 	-- Load personal logistic slots
-	for i, slot in pairs(serialized) do
-		if slot ~= nil then
-			player.set_personal_logistic_slot(tonumber(i), slot)
+	if v2_logistic_api then
+		local logistic_point = player.get_requester_point()
+
+		-- Remove old sections up to section_count
+		for i = logistic_point.sections_count, 1, -1 do
+			logistic_point.remove_section(i)
+		end
+
+		-- If this is an array instead of a table, migrate to v2 format
+		if serialized[1] ~= nil then
+			local section = logistic_point.add_section()
+			for i, slot in pairs(serialized) do
+				section.set_slot(i, {
+					value = { name = slot.name },
```

Comment:
`quality` must be specified because the default is "any qauality" which will then complain about "Can't specify non zero request with non trivial item filter condition" when `slot.min > 0`